configuration/postgres/metadata.go

```diff
@@ -55,8 +62,13 @@ func (m *metadata) InitWithMetadata(meta map[string]string) error {
 		return fmt.Errorf("invalid table name '%s'. non-alphanumerics or upper cased table names are not supported", m.ConfigTable)
 	}
 
-	// Azure AD auth is supported for this component
-	err = m.PostgresAuthMetadata.InitWithMetadata(meta, true)
+	opts := pgauth.InitWithMetadataOpts{
+		AzureADEnabled: m.UseAzureAD,
+		AWSIAMEnabled:  m.UseAWSIAM,
```

Comment:
Always `true` here as it indicates where it _can_ be used
```suggestion
		AzureADEnabled: true,
		AWSIAMEnabled:  true,
```